docs/guides/test-isolation.md

```diff
@@ -0,0 +1,119 @@
+# Test Isolation
+
+Because MFTF is a framework for testing a highly customizable and ever changing application, MFTF tests need to be properly isolated.
+
+## What is test isolation?
+
+Test isolation refers to a test that does not leave behind any data or configuration changes in the Magento instance.
+
+An MFTF test is considered fully isolated if:
+
+1. It does not leave data behind.
+1. It does not leave Magento configured in a different state than when the test started.
```

Comment:
Content looks good, just need to fix numbering error.

Reply:
Fixed 👍

Reply:
In our devdocs build system, we number everything 1. and then the system converts it to the correct number.

Reply:
Didn't know. Saw numbering at a few places in this PR. Closing this comment in that case.